lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -270,7 +270,36 @@ class TargetReflectionContext
   const swift::reflection::TypeInfo *
   getTypeInfo(const swift::reflection::TypeRef *type_ref,
               swift::remote::TypeInfoProvider *provider) override {
-    return m_reflection_ctx.getTypeInfo(type_ref, provider);
+    if (!type_ref)
+      return nullptr;
+
+    Log *log(GetLog(LLDBLog::Types));
+    if (log && log->GetVerbose()) {
+      std::stringstream ss;
+      type_ref->dump(ss);
+      log->Printf("[TargetReflectionContext::getTypeInfo] Getting "
+                  "type info for typeref:\n%s",
+                  ss.str().c_str());
+    }
+
+    auto type_info = m_reflection_ctx.getTypeInfo(type_ref, provider);
+    if (!type_info) {
+      std::stringstream ss;
+      type_ref->dump(ss);
+      LLDB_LOGF(log,
+                "[TargetReflectionContext::getTypeInfo] Could not get "
+                "type info for typeref:\n%s",
+                ss.str().c_str());
+    }
+
+    if (log && log->GetVerbose()) {
+      std::stringstream ss;
+      type_info->dump(ss);
+      log->Printf("[TargetReflectionContext::getTypeInfo] Found "
```

Comment:
Why not use LLDB_LOGV here?

Reply:
There's no way to do both LLDB_LOG with V and F

Reply:
That's true, LLDB_LOGV uses llvm's Format instead of printf like LLDB_LOG.
But then the question becomes: Could you use Format specifiers here — or — should you add an LLDB_LOGVF macro?

Reply:
I think Format would be better here anyway because AFAIK it also accepts std::string and StringRef, so you don't need to copy into a `c_str()`.